File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/dependencies/Dependency_ViewModelTests.cs
/// <filename>
///     Dependency_ViewModelTests.cs
/// </filename>
using dependencies;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class Dependency_ViewModelTests
{
    private dependencies.Dependency_ViewModel sut;

    private dependencies.Dependency_ViewModelTestSetup testSetup;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testSetup = new dependencies.Dependency_ViewModelTestSetupImpl();
        this.testSetup.Init();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void ViewModel_Dependency_Invocation_given_when_LoadView_then_()
    {
        this.BuildSut();
        this.when_LoadView();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testSetup.BuildSut();
    }

    public virtual void when_LoadView()
    {
    }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/dependencies/Dependency_ViewModelTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "Dependency_ViewModelTestSetup.h"
#include "Dependency_ViewModel.h"
#include "Dependency_ViewModelTestSetupImpl.h"

namespace dependencies
{
  class Dependency_ViewModelTests : public testing::Test
  {
  public:
    std::shared_ptr<Dependency_ViewModel> sut;
    std::shared_ptr<Dependency_ViewModelTestSetup> testSetup;
    virtual void BuildSut();
    virtual void when_LoadView();
  protected:
    void SetUp() override;
  };
  void Dependency_ViewModelTests::SetUp()
  {
    this->testSetup = std::make_shared<Dependency_ViewModelTestSetupImpl>();
    this->testSetup->Init();
  }
  TEST_F(Dependency_ViewModelTests, ViewModel_Dependency_Invocation_given_when_LoadView_then_)
  {
    this->BuildSut();
    this->when_LoadView();
  }
  void Dependency_ViewModelTests::BuildSut()
  {
    this->sut = this->testSetup->BuildSut();
  }
  void Dependency_ViewModelTests::when_LoadView()
  {
  }
}
